### PR TITLE
Updated to not start the instance when running installers

### PR DESCRIPTION
**Pull request by ramonsmits**
- [x] Update samples for other core versions besides 6
- [x] ~~Document DisableInstallers for v7~~

### CODE REVIEW

**Comment by ramonsmits** [comment]
Looking at the code for versions before core v5 and it doesn't seem that you cannot create but not start an endpoint. @Particular/nservicebus-maintainers is that correct?

**Comment by ramonsmits** [comment]
@Particular/nservicebus-maintainers In V6.4 `DisableInstallers` was introduced

- https://docs.particular.net/nservicebus/operations/installers?version=core_6

What exactly is its use? I always thought installers were never run unless explicitly via `EnableInstallers()`.

**Comment by timbussmann** [comment]
> What exactly is its use? I always thought installers were never run unless explicitly via EnableInstallers().

@ramonsmits that is not the case. See the v5 version of the installer doco page https://docs.particular.net/nservicebus/operations/installers?version=core_5 which states:

> By default, installers are only run when the application starts with a debugger attached. This includes applications being debugged remotely.

**Comment by ramonsmits** [comment]
@timbussmann thanks for the clarification, so DisableInstallers results in never to run the installers also when debugging prior to 7 and from 7 onwards we do not invoke the installers during debugging. Is that correct? That doco is so very difficult to understand differences between major versions.

**Comment by timbussmann** [comment]
> DisableInstallers results in never to run the installers also when debugging

that's correct. Note that the `DisableInstallers` API was added in 6.4 IIRC.

> from 7 onwards we do not invoke the installers during debugging.

that's correct, unless you explicitly specify `EnableInstallers`

**Comment by ramonsmits** [comment]
@timbussmann As you already provided help, could you review?

File: Snippets/Core/Core_5/Installers.cs [hunk 1]
```diff
@@ -32,6 +33,9 @@ public static void Main(string[] args)
             if (runInstallers)
             {
                 busConfiguration.EnableInstallers();
+                // This will run the installers but not start the instance.
+                Bus.Create(busConfiguration);
+                return; // Exit application
```
Comment (timbussmann) [inline Snippets/Core/Core_5/Installers.cs]: could we maybe just remove this and get rid of the resharper warning comment at the top of the file? or use Environment.Exit or so?

File: Snippets/Core/Core_6/Installers.cs [hunk 2]
```diff
@@ -27,13 +28,17 @@ class SwitchInstallersWithCommandline
...
         {
             var runInstallers = args.Any(x => x.ToLower() == "/runInstallers");
 
             if (runInstallers)
             {
                 endpointConfiguration.EnableInstallers();
+                // This will run the installers but not start the instance.
+                await Endpoint.Create(endpointConfiguration)
+                    .ConfigureAwait(false);
+                return; // Exit application
```
Comment (timbussmann) [inline Snippets/Core/Core_6/Installers.cs]: same here as the comment above

**Pull request merged**